planner/core/planbuilder.go

```diff
@@ -2882,7 +2878,7 @@ func (b *PlanBuilder) buildAnalyzeIndex(as *ast.AnalyzeTableStmt, opts map[ast.A
 		if isPrimaryIndex(idxName) {
 			handleCols := BuildHandleColsForAnalyze(b.ctx, tblInfo, true, nil)
 			// Fast analyze use analyze column to solve int handle.
-			if handleCols != nil && handleCols.IsInt() && b.ctx.GetSessionVars().EnableFastAnalyze {
+			if handleCols != nil && handleCols.IsInt() {
```

Comment:
Does this change make sense?

Reply:
Introduced by https://github.com/pingcap/tidb/pull/23254

Reply:
Should we remove the whole part? Since if `fast-analyze` is removed, the condition here should always be `False`:
<img width="1247" alt="image" src="https://github.com/pingcap/tidb/assets/7499936/196478d7-cf35-4f49-8b6d-c3905ffbc388">

Reply:
If we removed the whole part. Then we can not pass the test:
```go
	// Issue15993
	store := testkit.CreateMockStore(t)
	tk := testkit.NewTestKit(t, store)
	tk.MustExec("set @@tidb_enable_fast_analyze=1;")
	tk.MustExec("set @@tidb_analyze_version = 1")
	tk.MustExec("use test")
	tk.MustExec("drop table if exists t0")
	tk.MustExec("CREATE TABLE t0(c0 INT PRIMARY KEY);")
	tk.MustExec("ANALYZE TABLE t0 INDEX PRIMARY;")
```

Reply:
We will take PRIMARY as a column and try to search it, then we will get a missing column error.

Reply:
See: [`1b67765` (#47502)](https://github.com/pingcap/tidb/pull/47502/commits/1b677655e351a4d3632a2f164adbfdd9b6b25607)

Now, we only handle the int primary key. It is not perfect. But it is better than before. To fix this problem out of this PR's topic. I will create an issue for it.